lib/optimizely/event_builder.rb

```diff
@@ -41,198 +42,202 @@ def ==(event)
-      #
-      # event_key - Event key representing the event which needs to be recorded.
-      # user_id - ID for user.
-      # attributes - Hash representing user attributes and values which need to be recorded.
-      # event_tags - Hash representing metadata associated with the event.
-      # experiment_variation_map - Map of experiment ID to the ID of the variation that the user is bucketed into.
-      #
-      # Returns event hash encapsulating the conversion event.
-
-      @params = {}
-      add_common_params(user_id, attributes)
-      add_conversion_event(event_key)
-      add_event_tags(event_tags)
-      add_layer_states(experiment_variation_map)
-      Event.new(:post, CONVERSION_EVENT_ENDPOINT, @params, POST_HEADERS)
-    end
-
-    private
-
-    def add_common_params(user_id, attributes)
-      super
-      @params['isGlobalHoldback'] = false
-    end
-
-    def add_project_id
-      @params['projectId'] = @config.project_id
-    end
+      experiment_key = experiment['key']
+      experiment_id = experiment['id']
 
-    def add_account_id
-      @params['accountId'] = @config.account_id
-    end
+      decision = [{
+        campaign_id: @config.experiment_key_map[experiment_key]['layerId'],
+        experiment_id: experiment_id,
+        variation_id: variation_id,
+      }]
+
+      event = [{
+        entity_id: @config.experiment_key_map[experiment_key]['layerId'],
+        timestamp: get_timestamp(),
+        key: ACTIVATE_EVENT_KEY,
+        uuid: get_uuid()
+      }]
+
+      impressionEventParams = {
+        decisions: decision,
+        events: event
+      }
 
-    def add_user_id(user_id)
-      @params['visitorId'] = user_id
+      impressionEventParams;
     end
 
-    def add_attributes(attributes)
-      @params['userFeatures'] = []
-
-      return if attributes.nil?
+    def get_conversion_event_params(event_key, event_tags, experiment_variation_map)
+      # Creates object of params specific to conversion events
+      #
+      # event_key -                +String+ Event key representing the event which needs to be recorded
+      # event_tags -               +Hash+ Values associated with the event.
+      # experiment_variation_map - +Hash+ Map of experiment IDs to bucketed variation IDs
+      #
+      # Returns +Hash+ Impression event params
 
-      attributes.keys.each do |attribute_key|
-        # Omit falsy attribute values
-        attribute_value = attributes[attribute_key]
-        next unless attribute_value
+      conversionEventParams = []
 
-        # Skip attributes not in the datafile
-        attribute_id = @config.get_attribute_id(attribute_key)
-        next unless attribute_id
+      experiment_variation_map.each do |experiment_id, variation_id|
 
-        feature = {
-          'id' => attribute_id,
-          'name' => attribute_key,
-          'type' => 'custom',
-          'value' => attribute_value,
-          'shouldIndex' => true,
+        decision = {
```

Comment:
The name of this local variable 'decision' creates confusion because we also have a property in the final object named 'decisions' and in another context there is also something called decision service in Optimizely. It would be helpful if this local var can be renamed, it's only a container that gathers all data and finally gets pushes to params. In PHP, I have named it single snapshot, as the array is eventually called 'snapshots'

Reply:
https://github.com/optimizely/node-sdk/blob/94f940cff1872ec1023fd9eb2ad4a3781d31d61d/lib/core/event_builder/index.js#L109